includes/woopay/class-woopay-session.php

```diff
@@ -216,6 +260,70 @@ private static function get_payload_from_cart_token() {
 		return null;
 	}
 
+	/**
+	 * Summary of get_init_session_request
+	 *
+	 * @return array
+	 */
+	public static function get_init_session_request() {
+		$user        = wp_get_current_user();
+		$customer_id = WC_Payments::get_customer_service()->get_customer_id_by_user_id( $user->ID );
+		if ( null === $customer_id ) {
+			// create customer.
+			$customer_data = WC_Payments_Customer_Service::map_customer_data( null, new WC_Customer( $user->ID ) );
+			$customer_id   = WC_Payments::get_customer_service()->create_customer_for_user( $user, $customer_data );
+		}
+
+		$account_id = WC_Payments::get_account_service()->get_stripe_account_id();
+
+		$store_logo = WC_Payments::get_gateway()->get_option( 'platform_checkout_store_logo' );
+
+		include_once WCPAY_ABSPATH . 'includes/compat/blocks/class-blocks-data-extractor.php';
+		$blocks_data_extractor = new Blocks_Data_Extractor();
+
+		// This uses the same logic as the Checkout block in hydrate_from_api to get the cart and checkout data.
+		$cart_data = rest_preload_api_request( [], '/wc/store/v1/cart' )['/wc/store/v1/cart']['body'];
+		add_filter( 'woocommerce_store_api_disable_nonce_check', '__return_true' );
+		$preloaded_checkout_data = rest_preload_api_request( [], '/wc/store/v1/checkout' );
+		remove_filter( 'woocommerce_store_api_disable_nonce_check', '__return_true' );
+		$checkout_data = isset( $preloaded_checkout_data['/wc/store/v1/checkout'] ) ? $preloaded_checkout_data['/wc/store/v1/checkout']['body'] : '';
```

Comment:
These 2 requests are already being preloaded and exposed to the frontend by the Checkout Block. We needed this because this used to be a separate endpoint, but now we are doing this at html render time, which is the same time the checkout Block does it. Do you know if this method is using the already existing preloaded response instead of ignoring it and getting a new response? If not, can we get what is already in the frontend and use it instead of reloading?

Reply:
The logic was changed on https://github.com/Automattic/woocommerce-payments/pull/6831/commits/1afcf02fdb1f67f21b94fbd1289b2d49ac6e1989 and the session data is not printed in the page anymore.